index.js

```diff
@@ -212,24 +210,11 @@ function inferTypeByKeyword (schema) {
   return schema.type
 }
 
-function hasOf (schema) {
-  if (!schema) { return false }
-  if ('anyOf' in schema || 'oneOf' in schema) { return true }
-
-  var objectKeys = Object.keys(schema)
-  for (var i = 0; i < objectKeys.length; i++) {
-    var value = schema[objectKeys[i]]
-    if (typeof value === 'object') {
-      if (hasOf(value)) { return true }
-    }
-  }
-
-  return false
-}
-
-function hasIf (schema) {
+function dependsOnAjv (schema) {
   const str = JSON.stringify(schema)
-  return /"if":{/.test(str) && /"then":{/.test(str)
+  return (/"if":{/.test(str) && /"then":{/.test(str)) ||
+    /"(anyOf|oneOf)":\[/.test(str) ||
+    /"const":/.test(str)
```

Comment:
@zekth here's the same what `hasOf`, `hasIf` and `hasConst` does, do you see any tricky cases?

Reply:
Can't the first condition be factorized by:
```js
/"if":{.*"then":{/.test(str)
```

Reply:
I tried run benchmark, seems there no impact almost at all

Reply:
I joined all regexp into one